### DISCUSSION TITLE
Get the shell setting to work

### DISCUSSION THREAD

**Discussion by justinfriebel** [zed-industries/zed#6521] [Feature Requests]
I can't get my editor/shell to work with zsh. I've looked at a few other's settings.json but can't find anything that works. This is my current setting:

```
    "shell": {
        "program": "zsh"
    },
```
I've tried `/bin/zsh` as the value too with no luck, any ideas?

**Comment by fdionisi** [comment]
Hey @justinfriebel, let me see if I can help. Can you elaborate on what's not working exactly? Is the terminal not working at all, or is it loading a different shell than zsh?

**Reply by justinfriebel** [reply]
It works but just stays on bash no matter what I do.

**Reply by fdionisi** [reply]
Great. So, if you open your external terminal app, `zsh` is loaded as your default shell, is that correct? If that's the case you should be able to avoid any configuration, or at most try:
```json
{
  "terminal": {
    "shell": "system"
  }
}
```

As additional troubleshooting, does `zsh` boot when manually running the command from the Zed terminal? Could you also check `which zsh` and see where the executable location is? It may be worth trying to use that path directly in Zed config.

I hope this helps 🤞

**Reply by justinfriebel** [reply]
I tried using `system` and restarting Zed, but my terminal is still using bash.

`which zsh` returns `/bin/zsh`, which is why I tried it initially :(

Running `zsh` does boot it and it works which is great!

It does seem like I'll need to boot zsh it every time unfortunately.

**Reply by fdionisi** [reply]
Uhm... strange. So your external terminal app loads `zsh` by default, but Zed doesn't. And I bet you already tried to use bare `zsh` instead of its path or `system`.

I'm out of ideas. Hopefully someone else will jump in here.🤞

Meanwhile, good luck. And let me know if you find a solution. I'm pretty interested in what's the issue.

**Comment by Musheer360** [comment]
Use this:

```
"terminal": {
    "shell": {
      "program": "zsh"
    }
}
```

**Comment by kergoth** [comment]
I have the same issue, trying to set the shell in the project settings. It has no effect at all, keeps running bash.

**Comment by pckim93** [comment]
Same issue on Bluefin OS (Linux). It's not even bash, but it seems to be sh. It doesn't even read ~/.bashrc, it seems.

**Comment by neobytes247** [comment]
Sample that works for bash:
```
"terminal": {
    "shell": {
      "program": "/opt/homebrew/bin/bash"
    }
  }
```

**Comment by rtgiskard** [comment]
Any update? it's always `sh` in the container, even the user shell and env SHELL is bash

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
